modules/display/display.cpp

```diff
@@ -0,0 +1,357 @@
+#define DISPLAY_IR_DISPLAY_CONTROL_CURSOR_ON   0b00000010
+#define DISPLAY_IR_DISPLAY_CONTROL_CURSOR_OFF  0b00000000
+#define DISPLAY_IR_DISPLAY_CONTROL_BLINK_ON    0b00000001
+#define DISPLAY_IR_DISPLAY_CONTROL_BLINK_OFF   0b00000000
+
+#define DISPLAY_IR_FUNCTION_SET_8BITS    0b00010000
+#define DISPLAY_IR_FUNCTION_SET_4BITS    0b00000000
+#define DISPLAY_IR_FUNCTION_SET_2LINES   0b00001000
+#define DISPLAY_IR_FUNCTION_SET_1LINE    0b00000000
+#define DISPLAY_IR_FUNCTION_SET_5x10DOTS 0b00000100
+#define DISPLAY_IR_FUNCTION_SET_5x8DOTS  0b00000000
+
+#define DISPLAY_20x4_LINE1_FIRST_CHARACTER_ADDRESS 0
+#define DISPLAY_20x4_LINE2_FIRST_CHARACTER_ADDRESS 64
+#define DISPLAY_20x4_LINE3_FIRST_CHARACTER_ADDRESS 20
+#define DISPLAY_20x4_LINE4_FIRST_CHARACTER_ADDRESS 84
+
+#define DISPLAY_RS_INSTRUCTION 0
+#define DISPLAY_RS_DATA        1
+
+#define DISPLAY_RW_WRITE 0
+#define DISPLAY_RW_READ  1
+
+#define DISPLAY_PIN_RS  4
+#define DISPLAY_PIN_RW  5
+#define DISPLAY_PIN_EN  6
+#define DISPLAY_PIN_D0  7  
+#define DISPLAY_PIN_D1  8  
+#define DISPLAY_PIN_D2  9  
+#define DISPLAY_PIN_D3 10
+#define DISPLAY_PIN_D4 11
+#define DISPLAY_PIN_D5 12 
+#define DISPLAY_PIN_D6 13 
+#define DISPLAY_PIN_D7 14 
+
+#define DISPLAY_PIN_A_PCF8574 3
+
+#define I2C1_SDA PB_9
+#define I2C1_SCL PB_8
+
+#define PCF8574_I2C_BUS_8BIT_WRITE_ADDRESS 78
+
+//=====[Declaration of private data types]=====================================
+
+typedef struct{
+    int address;
+    char data;
+    bool displayPinRs;
+    bool displayPinRw; 
+    bool displayPinEn;
+    bool displayPinA;
+    bool displayPinD4;
+    bool displayPinD5;
+    bool displayPinD6;
+    bool displayPinD7;
+} pcf8574_t;
+
+//=====[Declaration and initialization of public global objects]===============
+
+DigitalOut displayD0( D0 );
+DigitalOut displayD1( D1 );
+DigitalOut displayD2( D2 );
+DigitalOut displayD3( D3 );
+DigitalOut displayD4( D4 );
+DigitalOut displayD5( D5 );
+DigitalOut displayD6( D6 );
+DigitalOut displayD7( D7 );
+DigitalOut displayRs( D8 );
+DigitalOut displayEn( D9 );
+
+I2C i2cPcf8574( I2C1_SDA, I2C1_SCL ); 
+
+//=====[Declaration of external public global variables]=======================
+
+//=====[Declaration and initialization of public global variables]=============
+
+//=====[Declaration and initialization of private global variables]============
+
+static display_t display;
+static pcf8574_t pcf8574;
+static bool initial8BitCommunicationIsCompleted;
+
+//=====[Declarations (prototypes) of private functions]========================
+
+static void displayPinWrite( uint8_t pinName, int value );
+static void displayDataBusWrite( uint8_t dataByte );
+static void displayCodeWrite( bool type, uint8_t dataBus );
+
+//=====[Implementations of public functions]===================================
+
+void displayInit( displayConnection_t connection )
+{
+    display.connection = connection;
+    
+    if( display.connection == DISPLAY_CONNECTION_I2C_PCF8574_IO_EXPANDER) {
```

Comment:
Cómo funciona esto ?
Recién lo entendí cuando llegué al final.
`//` is your friend.
En vez de poner un banner inútil explicándome que la definición de una función que no tiene un static adelante es la definición de una función pública, podrías explicar qué hace el código, que es lo que quien mira el archivo no sabe...